RecFCCeeCalorimeter/src/components/AugmentClustersFCCee.cpp

```diff
@@ -177,48 +283,151 @@ StatusCode AugmentClustersFCCee::execute([[maybe_unused]] const EventContext &ev
+        }
+
         if (m_thetaRecalcLayerWeights[k][layer]<0)
           sumThetaLayer[layer+startPositionToFill] += (eCell * theta);
         else
           sumThetaLayer[layer+startPositionToFill] += (weightLog * theta);
         sumWeightLayer[layer+startPositionToFill] += weightLog;
         sumPhiLayer[layer+startPositionToFill] += (eCell * phi);
-      }
+
+        // do pi0/photon shape var only for EMB
+        if (m_do_pi0_photon_shapeVar && systemID == 4) {
+          // E, theta_id, and module_id of cells in layer
+          vec_E_cell_layer[layer+startPositionToFill].push_back(eCell);
+          vec_theta_cell_layer[layer+startPositionToFill].push_back(theta_id);
+          vec_module_cell_layer[layer+startPositionToFill].push_back(module_id);
+          // sum them for width calculation
+          theta2_E_layer[layer+startPositionToFill] += theta_id * theta_id * eCell;
+          theta_E_layer[layer+startPositionToFill] += theta_id * eCell;
+          module2_E_layer[layer+startPositionToFill] += module_id * module_id * eCell;
+          module_E_layer[layer+startPositionToFill] += module_id * eCell;
+        }
+      }  // end of loop over cells
+    }  // end of loop over each system / readout
+
+    std::vector<std::pair<std::vector<int>, std::vector<double>>> theta_E_pair;
+    // local maxima (could be more than one) and the corresponding theta
+    std::vector<std::vector<double>> local_E_Max(numLayersTotal, std::vector<double>());
+    std::vector<std::vector<int>> local_E_Max_theta(numLayersTotal, std::vector<int>());
+
+    std::vector<double> E_cell_Max(numLayersTotal, 0.);
+    std::vector<double> E_cell_secMax(numLayersTotal, 0.);
+    std::vector<double> E_cell_Max_theta(numLayersTotal, 0.);
+    std::vector<double> E_cell_secMax_theta(numLayersTotal, 0.);
+    std::vector<double> E_cell_Min(numLayersTotal, std::numeric_limits<double>::max());
+
+    startPositionToFill = 0;
+    for (size_t k = 0; k < m_readoutNames.size(); k++) {
+      if (!m_do_pi0_photon_shapeVar)    break;
+      if (m_systemIDs[k] != 4)    continue;    // do pi0/photon shape var only for EMB
+      if (k > 0)    startPositionToFill += m_numLayers[k - 1];
+      // loop over layers
+      for (unsigned layer = 0; layer < m_numLayers[k]; layer++) {
+        // in case there's no cell in this layer (sometimes in layer 0)
+        if (vec_E_cell_layer[layer+startPositionToFill].empty()) {
+          vec_E_cell_layer[layer+startPositionToFill].push_back(0);
+          vec_theta_cell_layer[layer+startPositionToFill].push_back(0);
+          vec_module_cell_layer[layer+startPositionToFill].push_back(0);
+        }
+        auto result = MergeSumAndSort(vec_theta_cell_layer[layer+startPositionToFill], vec_E_cell_layer[layer+startPositionToFill]);
+        theta_E_pair.push_back(result);
+
+        // loop over theta IDs
+        for (size_t i = 0; i < theta_E_pair[layer+startPositionToFill].second.size(); i ++) {
```

Comment:
I think the part that finds the local maxima and the minimum between them is not behaving properly when there are some theta bins with 0 energy, since the corresponding cells are not attached to the cluster. One should take that into account

Reply:
OK, so will do the loop over cells instead

Reply:
Filled zero energy cells in 1D theta-E and module-E profiles